vault/forwarded_writer_oss.go

```diff
@@ -0,0 +1,112 @@
+//go:build !enterprise
+
+package vault
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/vault/sdk/logical"
+)
+
+// Our forwarded writer has two components: a reference to Core, allowing
+// us to tap into the GRPC client and resolved paths, and lower storage
+// layer to call upon when we don't wish to forward our writes.
+//
+// This implementation lives in OSS: while the GRPC connection isn't present
+// on OSS, we need to ensure paths written to these forwarded nodes correctly
+// template {{clusterId}} if they are later upgraded to Enterprise, and don't
+// just write with the template sentinel still there.
+//
+// XXX: In the future, we'll need to support wrapping transactional storage.
+type ForwardedWriter struct {
+	core      *Core
+	lower     logical.Storage
+	clusterID string
+}
+
+func (c *Core) NewForwardedWriter(ctx context.Context, wrapped logical.Storage, local bool) (logical.Storage, error) {
```

Comment:
Shouldn't we make use of `local`?

Reply:
Oh that's only for ent.  Maybe a comment to that effect is warranted, so no one attempts to "fix" it based on only look at the OSS repo?

Reply:
Ah, yes. Will do.

Reply:
Pushed.